reark/src/main/java/io/reark/reark/data/stores/cores/ContentProviderStoreCoreBase.java

```diff
@@ -298,27 +296,31 @@ private U mergedItem(@NonNull final U currentItem, @NonNull final U newItem) {
     }
 
     @NonNull
-    private Single<Boolean> createModifyingOperation(@NonNull final Func1<Subject<Boolean, Boolean>, CoreValue<U>> valueFunc) {
+    private Single<Boolean> createModifyingOperation(@NonNull final Function<Subject<Boolean>, CoreValue<U>> valueFunc) {
         BehaviorSubject<Boolean> completionNotifier = BehaviorSubject.create();
-        operationSubject.onNext(valueFunc.call(completionNotifier));
 
-        return completionNotifier.first()
-                .toSingle();
+        try {
+            operationSubject.onNext(valueFunc.apply(completionNotifier));
+        } catch (Exception e) {
+            Log.e(TAG, "Error creating operation!", e);
```

Comment:
Shouldn't you push the error to the subject?

Reply:
The apply call can't really throw, the try-catch is just to satisfy the Function method signature. I seem to recall there was a non-throwing version of the Rx functional interfaces but can't now find anything